staging/src/k8s.io/apiserver/pkg/endpoints/handlers/get.go

```diff
@@ -207,18 +207,26 @@ func ListResource(r rest.Lister, rw rest.Watcher, scope RequestScope, forceWatch
 		}
 
 		if hasName {
+			// FIXME: Check if namespaced resources have namespace set.
+			// TODO: How to check if the resource is namespaced here?
```

Comment:
@liggitt - how can we check here is the resource is namespaced or cluster-scoped?

Reply:
I'm actually rethinking whether we need to do that check... this works today:

```sh
$ kubectl get serviceaccounts --all-namespaces --field-selector=metadata.name=default --v=6
I0513 21:12:31.976525   16635 round_trippers.go:405] GET https://localhost:6443/api/v1/serviceaccounts?fieldSelector=metadata.name%3Ddefault&limit=500 200 OK in 10 milliseconds
NAMESPACE     NAME      SECRETS   AGE
default       default   1         14m
kube-public   default   1         14m
kube-system   default   1         14m
```

Reply:
let's ensure we have test coverage of the following API calls:
* `CoreV1().Secrets(NamespaceAll).List("metadata.name=foo")` only returns secrets named "foo"
* `CoreV1().Secrets(NamespaceAll).List("metadata.name=foo,metadata.namespace=bar")` only returns the secret named "foo" in the "bar" namespace
* `CoreV1().Secrets("bar").List("metadata.name=foo,metadata.namespace=bar")` only returns the secret named "foo" in the "bar" namespace
* `CoreV1().Secrets("bar").List("metadata.name=foo,metadata.namespace=baz")` returns no secrets (namespace field selector doesn't match namespace in path)

Reply:
OK. So I left this code as it was and added integration test to check those things.